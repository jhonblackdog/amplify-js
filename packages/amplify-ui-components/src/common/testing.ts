export const pixelThreshold = 10;
