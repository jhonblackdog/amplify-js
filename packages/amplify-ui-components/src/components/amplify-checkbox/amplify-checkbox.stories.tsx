import { storiesOf } from '@storybook/html';
import { text as textKnob } from '@storybook/addon-knobs';

const checkboxStories = storiesOf('amplify-checkbox', module);

checkboxStories.add('with label', () => {
  const label = textKnob('Label', 'Boise');
  return `<amplify-checkbox label=${label}></amplify-checkbox>`;
});
