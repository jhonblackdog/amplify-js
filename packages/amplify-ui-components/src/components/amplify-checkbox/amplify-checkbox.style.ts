import { css } from 'emotion';

export const checkbox = css`
  margin-bottom: 22px;
`;