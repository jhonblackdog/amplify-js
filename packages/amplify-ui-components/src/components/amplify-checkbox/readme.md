# amplify-checkbox



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type                                                               | Default     |
| --------------- | ---------------- | ----------- | ------------------------------------------------------------------ | ----------- |
| `inputProps`    | --               |             | `{ type: string; name?: string; value?: any; checked?: boolean; }` | `undefined` |
| `styleOverride` | `style-override` |             | `boolean`                                                          | `false`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
