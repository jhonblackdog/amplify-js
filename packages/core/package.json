{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
