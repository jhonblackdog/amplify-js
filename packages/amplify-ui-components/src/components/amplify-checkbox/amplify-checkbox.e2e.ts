import { newE2EPage } from '@stencil/core/testing';

describe('Checkbox Component:', () => {
  it('should render a amplify-checkbox', async () => {
    const page = await newE2EPage();
    await page.setContent(`<amplify-checkbox></amplify-checkbox>`);
    const el = await page.find('amplify-checkbox');
    expect(el).not.toBeNull();
  });
});