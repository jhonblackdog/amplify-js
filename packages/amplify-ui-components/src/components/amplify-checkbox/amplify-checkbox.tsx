import { Component, Prop, h } from '@stencil/core';
import { checkbox } from './amplify-checkbox.style';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_PREFIX } from '../../common/constants';

const STATIC_CHECKBOX_CLASS_NAME = `${AMPLIFY_UI_PREFIX}--checkbox`;

@Component({
  tag: 'amplify-checkbox',
  shadow: false,
})
export class AmplifyCheckbox {
  /** (Optional) Overrides default styling */
  @Prop() overrideStyle: boolean = false;
  /** Name of the checkbox */
  @Prop() name?: string;
  /** Value of the checkbox */
  @Prop() value?: string;
  /** Field ID used for the 'htmlFor' in the label */
  @Prop() fieldId: string;
  /** Label for the checkbox */
  @Prop() label: string;
  /** If `true`, the checkbox is selected. */
  @Prop() checked: boolean = false;
  /** If `true`, the checkbox is disabled */
  @Prop() disabled: boolean = false;

  private onClick = () => {
    this.checked = !this.checked;
  }

  render() {
    return (
      <span class={styleNuker(this.overrideStyle, STATIC_CHECKBOX_CLASS_NAME, checkbox)}>
        <input
          onClick={this.onClick}
          type="checkbox"
          name={this.name}
          value={this.value}
          id={this.fieldId}
          checked={this.checked}
          disabled={this.disabled}
        />
        <amplify-label htmlFor={this.fieldId}>{this.label}</amplify-label>
      </span>
    );
  }
}