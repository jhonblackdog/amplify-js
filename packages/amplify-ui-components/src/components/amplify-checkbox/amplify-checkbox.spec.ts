import { newSpecPage } from '@stencil/core/testing';
import { AmplifyCheckbox } from './amplify-checkbox';

describe('amplify-checkbox:', () => {
  describe('Default cases ->', () => {
    describe('Class logic ->', () => {
      let checkbox;

      beforeEach(() => {
        checkbox = new AmplifyCheckbox();
      });

      it('should have checked prop set to false', () => {
        expect(checkbox.checked).toBe(false);
      });

      it('should have disabled prop set to false', () => {
        expect(checkbox.disabled).toBe(false);
      });

      it('should have style override prop set to false', () => {
        expect(checkbox.styleOverride).toBe(false);
      });
    });
    describe('HTML logic ->', () => {
      it('should render with only a class of `amplify-ui--checkbox` when style override is true', async () => {
        const page = await newSpecPage({
          components: [AmplifyCheckbox],
          html: `<amplify-checkbox style-override='true'></amplify-checkbox>`
        });

        expect(page.root).toMatchSnapshot();
      });

      it('should render a checkbox with an input type of checkbox and label', async () => {
        const page = await newSpecPage({
          components: [AmplifyCheckbox],
          html: `<amplify-checkbox label="Seattle"></amplify-checkbox>`
        });

        expect(page.root).toMatchSnapshot();
      });

      it('should render a checkbox with an input type of checkbox and empty label', async () => {
        const page = await newSpecPage({
          components: [AmplifyCheckbox],
          html: `<amplify-checkbox></amplify-checkbox>`
        });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});