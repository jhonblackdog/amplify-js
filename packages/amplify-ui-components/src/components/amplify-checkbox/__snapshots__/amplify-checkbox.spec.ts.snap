// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-checkbox spec: Render logic -> should render a checkbox with an input type of checkbox and empty label by default 1`] = `
<amplify-checkbox>
  <span class="amplify-ui--checkbox css-1teaxst">
    <input type="checkbox">
    <amplify-label></amplify-label>
  </span>
</amplify-checkbox>
`;

exports[`amplify-checkbox spec: Render logic -> should render a checkbox with an input type of checkbox and label 1`] = `
<amplify-checkbox label="Seattle">
  <span class="amplify-ui--checkbox css-1teaxst">
    <input type="checkbox">
    <amplify-label>
      Seattle
    </amplify-label>
  </span>
</amplify-checkbox>
`;

exports[`amplify-checkbox spec: Render logic -> should render with only a class of \`amplify-ui--checkbox\` when style override is true 1`] = `
<amplify-checkbox override-style="true">
  <span class="amplify-ui--checkbox">
    <input type="checkbox">
    <amplify-label></amplify-label>
  </span>
</amplify-checkbox>
`;
